src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java

```diff
@@ -277,22 +277,36 @@ private boolean isUnusedImport(String imprt) {
     private void processIdent(DetailAST ast) {
         final DetailAST parent = ast.getParent();
         final int parentType = parent.getType();
+        final boolean isTypeDeclaration = TokenUtil.isTypeDeclaration(parentType);
+        final boolean isClassOrMethod = parentType == TokenTypes.DOT
+                || parentType == TokenTypes.METHOD_DEF || parentType == TokenTypes.METHOD_REF;
 
-        final boolean isPossibleDotClassOrInMethod = parentType == TokenTypes.DOT
-                || parentType == TokenTypes.METHOD_DEF;
-
-        final boolean isQualifiedIdent = parentType == TokenTypes.DOT
-                && !TokenUtil.isOfType(ast.getPreviousSibling(), TokenTypes.DOT)
-                && ast.getNextSibling() != null;
-
-        if (TokenUtil.isTypeDeclaration(parentType)) {
+        if (isTypeDeclaration) {
             currentFrame.addDeclaredType(ast.getText());
         }
-        else if (!isPossibleDotClassOrInMethod || isQualifiedIdent) {
+        else if (!isClassOrMethod || isQualifiedIdentifier(ast)) {
             currentFrame.addReferencedType(ast.getText());
         }
     }
 
+    /**
+     * Checks whether ast is a method reference usage.
```

Comment:
`Checks if given ast is a fully qualified identifier.`

Reply:
done